preload/single_download_late_used_preload.html

```diff
@@ -1,19 +1,20 @@
 <!DOCTYPE html>
+<title>Ensure preloaded resources are not downloaded again when used</title>
 <script src="/resources/testharness.js"></script>
 <script src="/resources/testharnessreport.js"></script>
-<script>
-    var t = async_test('Makes sure that preloaded resources are not downloaded again when used');
-</script>
-<div id=result></div>
 <link rel=preload href="resources/square.png" as=image>
 <script>
-    window.addEventListener("load", t.step_func(function() {
-        t.step_timeout(function() {
-            document.getElementById("result").innerHTML = "<image src='resources/square.png'>";
-            t.step_timeout(function() {
-                assert_equals(performance.getEntriesByType("resource").length, 3);
-                t.done();
-            }, 1000);
-        }, 1000);
-    }));
+    var link = document.getElementsByTagName("link")[0]
+    assert_equals(link.as, "image");
+    link.addEventListener("load", () => {
```

Comment:
I think the easiest fix is just to make the image slow loading. Otherwise we need to insert the `<link>` with js to be totally sure it can't load before the event handler is attached.